app/leave-app/backend/service.bal

```diff
@@ -0,0 +1,736 @@
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+import leave_service.authorization;
+import leave_service.database;
+import leave_service.email;
+import leave_service.employee;
+
+import ballerina/http;
+import ballerina/log;
+import ballerina/time;
+
+@display {
+    label: "Leave Backend Service",
+    id: "people-ops/leave-application"
+}
+service http:InterceptableService / on new http:Listener(9090) {
+
+    # Request interceptor.
+    # + return - authorization:JwtInterceptor
+    public function createInterceptors() returns http:Interceptor[] {
+        return [new authorization:JwtInterceptor()];
+    }
+
+    function init() returns error? {
+        log:printInfo("Leave application backend service started.");
+    }
+
+    # Get leaves for the given filters.
+    #
+    # + ctx - HTTP request context
+    # + req - HTTP request
+    # + email - Email of the user to filter the leaves
+    # + startDate - Start date filter
+    # + endDate - End date filter
+    # + isActive - Whether to filter active or inactive leaves
+    # + return - Return list of leaves
+    resource function get leaves(
+            http:RequestContext ctx,
+            http:Request req,
+            string? email = (),
+            string? startDate = (),
+            string? endDate = (),
+            boolean? isActive = (),
+            int? 'limit = (),
+            int? offset = 0
+    ) returns FetchedLeavesRecord|http:Forbidden|http:InternalServerError|http:BadRequest {
+
+        if email is string && !email.matches(WSO2_EMAIL_PATTERN) {
+            return <http:BadRequest>{
+                body: {
+                    message: string `${ERR_MSG_INVALID_WSO2_EMAIL} ${email}`
+                }
+            };
+        }
+
+        do {
+            readonly & authorization:CustomJwtPayload userInfo = check ctx.getWithType(authorization:HEADER_USER_INFO);
+            string|error jwt = req.getHeader(authorization:JWT_ASSERTION_HEADER);
+            if jwt is error {
+                fail error(ERR_MSG_NO_JWT_TOKEN_PRESENT, jwt);
+            }
+            if email != userInfo.email {
+                boolean validateForSingleRole = authorization:validateForSingleRole(userInfo, authorization:adminRoles);
+                if !validateForSingleRole {
+                    log:printWarn(string `The user ${userInfo.email} was not privileged to access the${false ? " admin " : " "}resource /leaves with email=${email.toString()}`);
```

Comment:
Exceeding 120 character length. Shall we fix other places also?

Reply:
Line 78 is a string. Are you asking to make it two lines?

Reply:
```
log:printWarn(string `The user ${userInfo.email} was not privileged to access the${
    false ? " admin " : " "}resource /leaves with email=${email.toString()}`);
```
yes, there's no harm breaking it as above